src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -111,7 +113,28 @@ std::shared_ptr<MulticlassLabels> CARTree::apply_multiclass(std::shared_ptr<Feat
 	auto current=get_root()->as<bnode_t>();
 
 	require(current, "Tree machine not yet trained.");
-	return apply_from_current_node(data->as<DenseFeatures<float64_t>>(), current)->as<MulticlassLabels>();
+	if (auto subfeat_data =
+	        std::dynamic_pointer_cast<DenseSubSamplesFeatures<float64_t>>(data))
+	{
+		size_t num_feat = subfeat_data->get_dim_feature_space();
+		auto num_vec = subfeat_data->get_num_vectors();
+		SGMatrix<float64_t> feature_matrix(num_feat, num_vec);
+		for (size_t i = 0; i < num_vec; i++)
+		{
+			const auto& v = subfeat_data->get_computed_dot_feature_vector(i);
```

Comment:
not sure why you are capture a reference to a temporary value here, but seems like it works?

Reply:
oh well, this is valid C++! Future reference https://blog.galowicz.de/2016/03/23/const_reference_to_temporary_object/

Reply:
The lifetime of a temporary object can be extended by binding to a const lvalue reference :D, i learn it from [this](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/).